src/main/java/com/example/demo/GreetingsController.java

```diff
@@ -0,0 +1,19 @@
+package com.example.demo;
+
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class GreetingsController {
+
+    @GetMapping("/greetingMood")
+   public Greeting Greetings(@RequestParam int mood) {  
```

Comment:
@RequestParam should always contain the name of the parameter. Example: @RequestParam("mood")

The reason is that a Java build in "release" mode will strip the parameter name information, thus preventing Spring-Boot from knowing that the argument is named "mood". By explicitly adding the parameter name, a "release" build will still work.